[docs] Document binary handling for BYTES columns in the CockroachDB connector (#7730)
debezium/debezium: documentation/modules/ROOT/pages/connectors/cockroachdb.adoc

|`BYTES`, `BYTEA`, `BLOB`
|`BYTES`

> |The connector decodes the hex literal that the changefeed emits for binary values.

Reviewer: ```suggestion
|The connector decodes the hexadecimal literal that the changefeed emits for binary values.
```

> The schema type and representation follow the xref:cockroachdb-property-binary-handling-mode[`binary.handling.mode`] property; `BYTES` is the default.

Reviewer: ```suggestion
The schema type and representation depend on the setting for the xref:cockroachdb-property-binary-handling-mode[`binary.handling.mode`] property; `BYTES` is the default.
```

Set one of the following options: `shared`, `none`, `custom`.
Because changefeed-based snapshots in CockroachDB do not require table locks, the default and recommended setting is `none`.

|[[cockroachdb-property-binary-handling-mode]]<<cockroachdb-property-binary-handling-mode, `+binary.handling.mode+`>>
|`bytes`
|Specifies how binary (`BYTES`) columns are represented in change events.
Set one of the following options:

`bytes`:: Represents binary data as a byte array.

> `base64`:: Represents binary data as a base64-encoded string.

Reviewer: ```suggestion
`base64`:: Represents binary data as a Base64-encoded string.
```

> `base64-url-safe`:: Represents binary data as a base64-url-safe-encoded string.

Reviewer: ```suggestion
`base64-url-safe`:: Represents binary data as a URL-safe Base64-encoded string.
```

- _Base64_ is the established term and is typically capitalized because it refers to a specific encoding scheme.

- _URL-safe Base64_ is the common industry phrasing used in standards, libraries, and documentation.

- The current wording, _base64-url-safe-encoded_, reads like a chain of modifiers and is harder to parse.

- We should try to stick to familiar terminology and avoid complex compound modifiers.

> `hex`:: Represents binary data as a hex-encoded (base16) string.

Reviewer: ```suggestion

`hex`:: Represents binary data as a hexadecimal-encoded (Base16) string.
```
